api/graphql/resolvers/user.go

```diff
@@ -316,46 +321,61 @@ func (r *mutationResolver) UserRemoveRootAlbum(ctx context.Context, userID int,
 		}
 
 		// Cleanup if no user owns the album anymore
-		var userAlbumCount int
-		if err := tx.Raw("SELECT COUNT(user_id) FROM user_albums WHERE album_id = ?", albumID).Scan(&userAlbumCount).Error; err != nil {
+		deletedAlbumIDs, err = cleanup(tx, albumID, deletedAlbumIDs, childAlbumIDs)
+		if err != nil {
 			return err
 		}
 
-		if userAlbumCount == 0 {
-			deletedAlbumIDs = append(childAlbumIDs, albumID)
-			childAlbumIDs = nil
-
-			// Delete albums from database
-			if err := tx.Delete(&models.Album{}, "id IN (?)", deletedAlbumIDs).Error; err != nil {
-				deletedAlbumIDs = nil
-				return err
-			}
-		}
-
 		return nil
 	})
 
 	if transactionError != nil {
 		return nil, transactionError
 	}
 
+	if err := clearCacheAndReloadFaces(deletedAlbumIDs, db); err != nil {
+		return nil, err
+	}
+
+	return &album, nil
+}
+
+func cleanup(tx *gorm.DB, albumID int, deletedAlbumIDs []int, childAlbumIDs []int) ([]int, error) {
+	var userAlbumCount int
+	if err := tx.Raw("SELECT COUNT(user_id) FROM user_albums WHERE album_id = ?",
+		albumID).Scan(&userAlbumCount).Error; err != nil {
+
+		return nil, err
+	}
+
+	if userAlbumCount == 0 {
+		deletedAlbumIDs = append(childAlbumIDs, albumID)
+		childAlbumIDs = nil
+		// Delete albums from database
+		if err := tx.Delete(&models.Album{}, "id IN (?)", deletedAlbumIDs).Error; err != nil {
+			deletedAlbumIDs = nil
+			return nil, err
+		}
+	}
+	return deletedAlbumIDs, nil
+}
+
+func clearCacheAndReloadFaces(deletedAlbumIDs []int, db *gorm.DB) error {
```

Comment:
it's better to put `db` in the first parameter.

Reply:
done